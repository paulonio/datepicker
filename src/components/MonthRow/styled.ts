import styled from 'styled-components';

export const Month = styled.div`
  color: #333333;
  font-size: 14px;
  font-weight: 700;
  line-height: 19px;
`;

export const MonthWrapper = styled.div`
  padding: 5px 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`;

export const YearIconWrapper = styled.div`
  width: 10px;
  height: 10px;
  cursor: pointer;
`;
