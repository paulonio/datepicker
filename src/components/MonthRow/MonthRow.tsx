import React, { FC } from 'react';

import { useMonthChangeHandlers } from '@hooks/useMonthChangeHandlers';
import { MONTHS } from '@constants/constants';

import PrevYearIcon from '@icons/PrevYearIcon';
import PrevIcon from '@icons/PrevIcon';
import NextIcon from '@icons/NextIcon';
import NextYearIcon from '@icons/NextYearIcon';
import type { View } from '@/types/types';

import { IconWrapper, Month, MonthWrapper } from './styled';

interface MonthRowProps {
  type: View;
  date: Date;
  setDate: (date: Date) => void;
}

const MonthRow: FC<MonthRowProps> = ({ type, date, setDate }) => {
  const { handlePrevClick, handleNextClick, handleNextYearClick, handlePrevYearClick } =
    useMonthChangeHandlers(type, date, setDate);

  return (
    <MonthWrapper>
      <IconWrapper>
        <PrevYearIcon onClick={handlePrevYearClick} data-testid="prev-year" />
      </IconWrapper>
      <IconWrapper>
        <PrevIcon onClick={handlePrevClick} data-testid="prev" />
      </IconWrapper>
      <Month data-testid="month-row">
        {MONTHS[date.getMonth()]} {`${date.getFullYear()}`}
      </Month>
      <IconWrapper>
        <NextIcon onClick={handleNextClick} data-testid="next" />
      </IconWrapper>
      <IconWrapper>
        <NextYearIcon onClick={handleNextYearClick} data-testid="next-year" />
      </IconWrapper>
    </MonthWrapper>
  );
};

export default MonthRow;
