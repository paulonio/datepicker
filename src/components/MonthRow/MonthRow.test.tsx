import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import MonthRow from '@components/MonthRow/MonthRow';

describe('MonthRow', () => {
  const date = new Date(2023, 4, 11);

  it('should render MonthRow', () => {
    const { container } = render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    expect(container).toMatchSnapshot();
  });

  it('should show correct month', () => {
    render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    const monthRow = screen.getByTestId('month-row');
    expect(monthRow).toHaveTextContent('May 2023');
  });

  it('should correctly set prev month', () => {
    render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    const prevButton = screen.getByTestId('prev');
    const monthRow = screen.getByTestId('month-row');
    fireEvent.click(prevButton);
    waitFor(async () => {
      await expect(monthRow).toHaveTextContent('April 2023');
    });
  });

  it('should correctly set prev year', () => {
    render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    const prevYearButton = screen.getByTestId('prev-year');
    const monthRow = screen.getByTestId('month-row');
    fireEvent.click(prevYearButton);
    waitFor(async () => {
      await expect(monthRow).toHaveTextContent('May 2022');
    });
  });

  it('should correctly set next month', () => {
    render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    const nextButton = screen.getByTestId('next');
    const monthRow = screen.getByTestId('month-row');
    fireEvent.click(nextButton);
    waitFor(async () => {
      await expect(monthRow).toHaveTextContent('June 2023');
    });
  });

  it('should correctly set next year', () => {
    render(<MonthRow type="monthly" date={date} setDate={() => {}} />);

    const nextYearButton = screen.getByTestId('next-year');
    const monthRow = screen.getByTestId('month-row');
    fireEvent.click(nextYearButton);
    waitFor(async () => {
      await expect(monthRow).toHaveTextContent('May 2024');
    });
  });
});
