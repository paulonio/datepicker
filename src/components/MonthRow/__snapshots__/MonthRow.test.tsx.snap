// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthRow should render MonthRow 1`] = `
<div>
  <div
    class="sc-gTRrQi keXuBg"
  >
    <div
      class="sc-ewnqHT iHhGPq"
    >
      <svg
        data-testid="prev-year"
        fill="currentColor"
        focusable="false"
        height="10"
        role="img"
        viewBox="0 0 9 15"
        width="10"
      >
        <path
          d="M0 7.2L7.2 14.4L8.32 13.2L2.4 7.2L8.32 1.2L7.2 0L0 7.2Z"
        />
      </svg>
    </div>
    <div
      class="sc-ewnqHT iHhGPq"
    >
      <svg
        data-testid="prev"
        fill="currentColor"
        focusable="false"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M11.7266 12.5L12.6666 11.56L9.61329 8.5L12.6666 5.44L11.7266 4.5L7.72663 8.5L11.7266 12.5Z"
        />
        <path
          d="M7.33332 12.5L8.27332 11.56L5.21998 8.5L8.27331 5.44L7.33331 4.5L3.33332 8.5L7.33332 12.5Z"
        />
      </svg>
    </div>
    <div
      class="sc-bgqQcB fnzUFW"
      data-testid="month-row"
    >
      May
       
      2023
    </div>
    <div
      class="sc-ewnqHT iHhGPq"
    >
      <svg
        data-testid="next"
        fill="currentColor"
        focusable="false"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M4.27337 4.5L3.33337 5.44L6.38671 8.5L3.33337 11.56L4.27337 12.5L8.27337 8.5L4.27337 4.5Z"
        />
        <path
          d="M8.66668 4.5L7.72668 5.44L10.78 8.5L7.72668 11.56L8.66668 12.5L12.6667 8.5L8.66668 4.5Z"
        />
      </svg>
    </div>
    <div
      class="sc-ewnqHT iHhGPq"
    >
      <svg
        data-testid="next-year"
        fill="currentColor"
        focusable="false"
        height="10"
        role="img"
        viewBox="0 0 9 15"
        width="10"
      >
        <path
          d="M1.12 0L0 1.2L5.92 7.2L0 13.2L1.12 14.4L8.32 7.2L1.12 0Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
